app/models/product.rb

```diff
@@ -0,0 +1,14 @@
+class Product < ApplicationRecord
+  belongs_to :user
+  has_many :reviews, dependent: :destroy
+  has_many :cart_items, dependent: :destroy
+  has_many :order_items, dependent: :destroy
+  has_and_belongs_to_many :categories
+  has_many :carts, through: :cart_items
+  has_many :orders, through: :order_items
+
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :price, presence: true, numericality: { greater_than: 0 }
```

Comment:
Price can be zero. as we might give the user free item depending on offers.

Reply:
Okay bhai. Fixing it. Thank you.